sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeComposeScreen.kt

```diff
@@ -1,29 +1,64 @@
 package co.nimblehq.sample.compose.ui.screens.home
 
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxSize
+import android.widget.Toast
+import androidx.compose.foundation.layout.*
 import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Scaffold
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
 import co.nimblehq.sample.compose.R
+import co.nimblehq.sample.compose.lib.IsLoading
 import co.nimblehq.sample.compose.model.UiModel
+import co.nimblehq.sample.compose.ui.AppDestination
 import co.nimblehq.sample.compose.ui.screens.AppBar
-import co.nimblehq.sample.compose.ui.theme.ComposeSampleTheme
+import co.nimblehq.sample.compose.ui.theme.ComposeTheme
+import co.nimblehq.sample.compose.ui.userReadableMessage
 
 // TODO: Rename to 'HomeScreen'
 @Composable
 fun HomeComposeScreen(
+    viewModel: HomeComposeViewModel = hiltViewModel(),
+    navigator: (destination: AppDestination) -> Unit
+) {
+    val context = LocalContext.current
+    LaunchedEffect(Unit) {
+        viewModel.error.collect { error ->
+            val message = error.userReadableMessage(context)
+            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+        }
+    }
+    LaunchedEffect(viewModel) {
+        viewModel.navigator.collect { destination -> navigator(destination) }
+    }
+
+    val showLoading: IsLoading by viewModel.showLoading.collectAsState()
+    val uiModels: List<UiModel> by viewModel.uiModels.collectAsState()
+
+    HomeScreenContent(
+        uiModels = uiModels,
+        showLoading = showLoading,
+        onItemClick = viewModel::navigateToSecond
+    )
+}
+
+@Composable
+private fun HomeScreenContent(
     uiModels: List<UiModel>,
-    showLoading: Boolean,
+    showLoading: IsLoading,
```

Comment:
I have a question. Should we remember this `showLoading` flag to reduce the recomposition in case there is no changed value?
```
var isShowLoading by remember { mutableStateOf(true) }
LaunchedEffect(key1 = showLoading) {
        isShowLoading = showLoading
    }
```

Reply:
@lydiasama `showLoading` is managed by ViewModel by a Flow so we don't need to use `remember` to keep its state because it's a state already by using `collectAsState()`. Check out more detail here https://developer.android.com/jetpack/compose/state#use-other-types-of-state-in-jetpack-compose 👋

Reply:
@luongvo I see. So if we access it by `collectAsState()`, it will handle as `state` and we don't wanna concern about the recomposition. 👍🏻  Thank you for your explanation.